libs/api/domains/vehicles/src/lib/api-domains-vehicles-mileage.resolver.ts

```diff
@@ -59,23 +60,37 @@ export class VehiclesMileageResolver {
     nullable: true,
   })
   @Audit()
-  postVehicleMileageReading(
+  async postVehicleMileageReading(
     @Args('input') input: PostVehicleMileageInput,
     @CurrentUser() user: User,
   ) {
-    return this.vehiclesService.postMileageReading(user, input)
+    const res = await this.vehiclesService.postMileageReading(user, {
+      ...input,
+      mileage: Number(input.mileage ?? input.mileageNumber),
```

Comment:
I'd move out of the call and extend the logic a bit, something like

```
let mileage

if (input.mileageNumber) {
  mileage = input.mileageNumber
} else if (input.mileage) {
  mileage = Number.parseInt(input.mileage)
}

if (!mileage || Number.isNaN(mileage)) {
  // error
}
  
```

Reply:
Yeah, my initial though was to handle this differently.
But after thinking about this, why not let the service itself handle if there is a faulty input?

With the setup in the PR it will return an error if the input from `mileage` is anything other than a 'string as number' (eg. "ABC").
And if it is a correct string "123" it will pass.

Same with if there's no input for `input.mileage` it will get the input from `input.mileageNumber` and if that's incorrect it will return an error as well. Same as it would in the initial setup where we would just take the input from the mutation as is and let the service handle it.